src/model/bpmn/internal/types.ts

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2023 Bonitasoft S.A.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
```

Comment:
**issue:** 

I find the syntax and the handling of extensions on the Shape object difficult to understand.

Instead of using complex objects like the one shown below, why didn't we opt for simpler objects?
```
type ShapeExtensions = EdgeExtensions & {
	fill?: { color?: string; }
	[key in ShapeExtensions]: string;
};

type EdgeExtensions = {
	stroke?: { color?: string; }
	label?: { color?: string; }
	[key in EdgeExtensions]: string;
};
```

If the purpose is to prepare for the future extension system, we can use [Model augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation), similar to what is done in Material UI.

By using model augmentation, developers creating new extensions won't need to define entirely new types. They can simply override the declaration of the existing type, as demonstrated in the following Material UI examples:

- [Palette customization](https://mui.com/material-ui/customization/palette/#typescript)
- [Theming customization](https://mui.com/material-ui/customization/theming/#typescript)
- [Creating new component variants](https://mui.com/material-ui/customization/theme-components/#creating-new-component-variants)

Reply:
After discussions with @csouchet we agreed to amend this PR with the extension types proposed above.

Reply:
Done in d957d43